scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -307,16 +315,15 @@
   (pending-offers [this pool-name]
     (log/info "In" name "compute cluster, looking for offers for pool" pool-name)
     (let [timer (timers/start (metrics/timer "cc-pending-offers-compute" name))
-          pods (add-starting-pods this @all-pods-atom)
-          nodes @current-nodes-atom
-          node-name->pods (api/pods->node-name->pods pods)
-          offers-all-pools (generate-offers this nodes node-name->pods)
-          ; TODO: We are generating offers for every pool here, and filtering out only offers for this one pool.
-          ; TODO: We should be smarter here and generate once, then reuse for each pool, instead of generating for each pool each time and only keeping one
-          offers-this-pool (get offers-all-pools pool-name)
-          offers-this-pool-for-logging (into []
-                                             (map #(into {} (select-keys % [:hostname :resources]))
-                                                  offers-this-pool))]
+          pods (add-starting-pods this @all-pods-atom) ; Safe. O(#pods)
+          nodes @current-nodes-atom ; Safe. O(#nodes)
```

Comment:
The comments here don't seem too helpful to me.

Reply:
Deleted.